mlrun/execution.py

```diff
@@ -194,6 +194,11 @@ def artifacts(self):
         """Dictionary of artifacts (read-only)"""
         return deepcopy(self._artifacts_manager.artifact_list())
 
+    @property
+    def artifact_uris(self):
+        """Dictionary of artifact URIs (read-only)"""
+        return deepcopy(self._artifacts_manager.artifact_uris)
```

Comment:
No need for `deepcopy` for a dictionary when keys and values are strings

Reply:
why not? the dictionary can be altered

Reply:
I think she means you can use regular copy without deep but I think the intention is to provide read-only structure which means it is safer to use deepcopy if in the future the structure of artifact uris would be different